src/core/mutatus.scala

```diff
@@ -221,80 +260,107 @@ case class Dao[T](kind: String)(implicit svc: Service, namespace: Namespace) {
   }
 }
 
-/** generic type for Datastore datatypes */
-class DsType(val set: (Entity.Builder, String) => Entity.Builder)
-
-/** companion object for DsType */
-object DsType {
-  final case class DsString(value: String) extends DsType(_.set(_, value))
-  final case class DsLong(value: Long) extends DsType(_.set(_, value))
-  final case class DsBoolean(value: Boolean) extends DsType(_.set(_, value))
-  final case class DsDouble(value: Double) extends DsType(_.set(_, value))
-  final case class DsKey(value: Ref[_]) extends DsType(_.set(_, value.ref))
-  final case class DsLatLng(value: Geo) extends DsType(_.set(_, value.toLatLng))
-  final case object DsRemove extends DsType(_.remove(_))
-}
-
 /** companion object for `Decoder`, including Magnolia generic derivation */
 object Decoder extends Decoder_1 {
   type Typeclass[T] = Decoder[T]
 
   /** combines `Decoder`s for each parameter of the case class `T` into a `Decoder` for `T` */
-  def combine[T](caseClass: CaseClass[Decoder, T]): Decoder[T] =
-    (obj, prefix) =>
+  def combine[T](caseClass: CaseClass[Decoder, T]): Decoder[T] = {
+    case entityValue: EntityValue =>
+      val entity = entityValue.get()
       caseClass.construct { param =>
```

Comment:
I'm not sure if it helps, but Magnolia has a `constructMonadic` method now (it wasn't there when the original code was written) which will _traverse_ over the parameters of the case class. I think you could switch to `constructMonadic` and remove the `.get` at the end.